File: arrays/containsDuplicate.go
package main
/*

link problem leetcode -> https://leetcode.com/problems/contains-duplicate/
time complexity  -> O(n)
space complexity -> O(n)

*/
func containsDuplicate(nums []int) bool {
	if len(nums) == 0 || len(nums) == 1{
		return false
	}
    mymap := make(map[int]int)

	for _, v := range nums {
		if mymap[v] == 1{
			return true
		}
		mymap[v] += 1
	}

	return false
}

File: hashmaps_and_sets/contains_duplicate.py
#Link to problem: https://leetcode.com/problems/contains-duplicate/description/
#Time complexity: O(n)
#Space complexity: O(n)
                       
class Solution:
    def contains_duplicate(self,nums:list[int]) -> bool:
        map = {}

        for num in nums:
            value = map.get(num)
            print(value)
            if value is None:
                map[num] = 1
            else:
                
                if value + 1 >= 2:
                    return True
                map[num] += 1

        return False
                                        
solution = Solution()
print(solution.contains_duplicate([1,2,3,1]))
print(solution.contains_duplicate([1,2,3,4]))
